app/models/course/lecture.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+class Course::Lecture < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  has_many_attachments on: :content
+
+  belongs_to :course, inverse_of: :lectures
+
+  scope :sorted_by_date, -> { order(start_at: :desc) }
+  scope :sorted_by_sticky, -> { order(sticky: :desc) }
+
+  def to_partial_path
+    'course/announcements/announcement'
```

Comment:
For allowing future customization, it's better to use it's own partial instead of the announcement one. Otherwise later some guy changes announcement partial and it will surprise him that the lecture also get changed.  what do you think ?